test/integration/replicaset_helpers.go

```diff
@@ -0,0 +1,69 @@
+package integration
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	testutils "github.com/kong/gateway-operator/pkg/utils/test"
+
+	operatorv1beta1 "github.com/kong/kubernetes-configuration/api/gateway-operator/v1beta1"
+)
+
+// FindReplicaSetNewerThan finds a ReplicaSet created after or at the specified time.
+// This helper logs detailed information about ReplicaSets found to help
+// with debugging timestamp issues.
+//
+// It will return nil if no ReplicaSet is found or if there's an error listing ReplicaSets.
+// If multiple ReplicaSets are found that were created after or at the specified time,
+// it will fail the test immediately with t.FailNow().
+// A single matching ReplicaSet is required for the test to proceed.
```

Comment:
Mentioning test here, as in "for the test to proceed" seems a bit unclear as there's no test in this context. I suggest to reword this to:

```suggestion
// It succeed when exactly 1 matching ReplicaSet is found.
```